src/tests/smart_contract_test.ts

```diff
@@ -55,6 +56,121 @@ describe("SmartContract", () => {
     });
   });
 
+  describe("SmartContract.register", () => {
+    const networkId = erc20Model.network_id;
+    const contractName = erc20Model.contract_name;
+    const contractAddress = erc20Model.contract_address;
```

Comment:
Is the erc20Model fixture an external contract? Probably worth making an external contract for this (We can fast-follow with this though)

Reply:
Sure, I've added a new external model